src/redisraft.c

```diff
@@ -649,13 +649,16 @@ static void handleRedisCommandAppend(RedisRaftCtx *rr,
         RaftRedisCommand *cmd = cmds->commands[i];
         size_t cmdlen;
         const char *cmdstr = RedisModule_StringPtrLen(cmd->argv[0], &cmdlen);
+        /* skip multi */
+        if (i == 0 && cmdlen == 5 && !strncasecmp(cmdstr, "MULTI", 5)) {
+            continue;
+        }
 
         enterRedisModuleCall();
         RedisModuleCallReply *reply = RedisModule_Call(ctx, cmdstr, "DCEMv", cmd->argv + 1, cmd->argc - 1);
```

Comment:
Redis OSS documentation for Dry-Run mode: 
``` *     * 'D' -- A "Dry Run" mode. Return before executing the underlying call().
 *              If everything succeeded, it will return with a NULL, otherwise it will
 *              return with a CallReply object denoting the error, as if it was called with
 *              the 'E' code.
 ```

But it seems we will return an error reply only when calling with `E` flag. Need to change the documentation there.

Reply:
cofused